server/controllers/testsController.js

```diff
@@ -0,0 +1,193 @@
+const testsController = {};
+
+/*
+  Middleware to verify that we have the properties we are looking for in our input object. If it fails, return 400 error code.
+  verify the reqest body has all the relevant properties and save them to res.locals.header and res.locals.assertions
+
+  EXPECTED INPUT:
+    {
+      header: {
+                endpoint : '/',
+                method: 'POST'
+                req_body: {a:1} // Needed for POST and PATCH, optional for DELETE
+      },
+      assertions: [
+                    { content: '/text\/html/' },
+                    { status: 200 },
+                    { res_body: { a: 'b' } }
+                    ...
+      ]
+    }
+    EXPECTED OUTPUT: res.locals.header, res.locals.assertions
+*/
+testsController.verifyInput = (req, res, next) => {
+  try {
+    const { header, assertions } = req.body;
+
+    // Check for improper input
+    if (
+      typeof header.endpoint !== 'string' ||
+      typeof header.method !== 'string' ||
+      !Array.isArray(assertions)
+    ) {
+      throw new Error('Missing properties or wrong data types in request.');
+    }
+
+    // Check for correct request methods
+    header.method = header.method.toUpperCase();
+    if (
+      header.method !== 'GET' &&
+      header.method !== 'POST' &&
+      header.method !== 'PATCH' &&
+      header.method !== 'DELETE'
+    ) {
+      throw new Error('Invalid request method provided.');
+    }
+    console.log(header);
+    // POST and PATCH methods must have request body provided
+    if (
+      (header.method === 'POST' || header.method === 'PATCH') &&
+      !header.req_body
+    ) {
+      throw new Error('Body not provided with POST or PATCH request.');
+    }
+
+    res.locals.header = header;
+    res.locals.assertions = assertions;
+    return next();
+  } catch (err) {
+    return next({
+      log: 'verifyInput middleware error: ' + err,
+      message: { err }
+    });
+  }
+};
+
+/* Converts header part of input object into header of test code
+
+  EXPECTED INPUT: res.locals.header, res.locals.assertions
+  EXPECTED OUTPUT: res.locals.headerOutput = Array of Strings
+    [
+      `describe('/', () => {`,
+        `describe('GET', () => {`,
+          `it('responds with status 200 and content type text/html', () => request(server)`,
+            `.get('/')`
+      ]
+*/
+testsController.headerGenerator = (header, assertions) => {
```

Comment:
amazing idea in splitting the code so that you can test them in jest.  As mentioned before, you might consider not storing the helper compiler functions in testsController as they are not meant to be passed as middleware

Reply:
Will separate helper function out to separate file to then import